src/OrbitGl/EventTrack.cpp

```diff
@@ -44,7 +44,7 @@ void EventTrack::Draw(GlCanvas* canvas, PickingMode picking_mode, float z_offset
     return;
   }
 
-  Batcher* batcher = canvas->GetBatcher();
+  Batcher* ui_batcher = canvas->GetBatcher();
```

Comment:
Have you thought about passing the Batcher here as well, so the track does not need to know about where batchers are managed or how many batchers the canvas actually uses?

Do we even need to pass the Canvas here if we do it like this? I would like to get rid of the GlCanvas* parameter here, because it is used very weirdly: We pass the canvas in `Draw`, but then we store it in `canvas_` because some other methods rely on it. These kind of side effects make testing hard.

Reply:
Yes, very weirdly. The next step is passing Batcher y TextRenderer here instead of the Canvas, but I'm still thinking about how to solve this "strange storing of canvas".